File: cow/src/MyCode.java
public class MyCode extends vrml.node.Script
{
  private vrml.field.SFBool flag;
  private vrml.field.SFColor color_changed;

  public void initialize()
  {
    flag = (vrml.field.SFBool)getField("flag");
    color_changed = (vrml.field.SFColor)getEventOut("color_changed");
  }

  public void processEvent(vrml.Event event)
  {
    if (event.getName().equals("isActive"))
    {
      vrml.field.ConstSFBool isActive = (vrml.field.ConstSFBool)event.getValue();
      //System.out.println("isActive(" + isActive.getValue() + ")"); // Debug output
      if (isActive.getValue() == true)
      {
        if (flag.getValue() == false)
        {
          flag.setValue(true);
          //System.out.println("color is green"); // Debug output
          color_changed.setValue(0, 1, 0);
        }
        else
        {
          flag.setValue(false);
          //System.out.println("color is red"); // Debug output
          color_changed.setValue(1, 0, 0);
        }
      }
    }
  }
}

File: IR_VRML/IR/instantreality.org-tutorials/instantreality.org-tutorials/implementing-script-nodes-in-javascript/MyCode.js
function isActive(value)
{
  //print('isActive(' + value + ')'); // Debug output
  if (value == true)
  {
    if (flag == false)
    {
      flag = true;
      //print('color is green'); // Debug output
      color_changed = new SFColor(0, 1, 0);
    }
    else
    {
      flag = false;
      //print('color is red'); // Debug output
      color_changed = new SFColor(1, 0, 0);
    }
  }
}
